File: game/tv_show/going_cave.py
from game_data import GameData
from phone_events import PhoneEvents
from tv_show.attract import Attract
from tv_show.in_cave import InCave
from tv_show.tv_show_resources import TvShowResources
from tv_show.video_state import VideoState


class GoingCave(VideoState):
    def __init__(self, context: GameData):
        super().__init__(context, TvShowResources.videos_going_scylla[context.country])

    def process_events(self, phone_events: PhoneEvents):
        super().process_events(phone_events)

        if phone_events.hungup:
            return Attract

        if self.video_ended():
            return InCave

        return None

File: game-web/src/tvShow/goingCave.ts
/** Port of game/tv_show/going_cave.py */

import { type StateClass } from '../state';
import { TvShowResources } from './tvShowResources';
import { VideoState } from './videoState';
import { Attract } from './attract';
import { InCave } from './inCave';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class GoingCave extends VideoState {
  constructor(context: GameData) {
    super(
      context,
      TvShowResources.videosGoingScylla[context.country],
      false,
      TvShowResources.audioGoingScylla[context.country],
    );
  }

  processEvents(events: PhoneEvents): StateClass | null {
    super.processEvents(events);
    if (events.hungup)     return Attract;
    if (this.videoEnded()) return InCave;
    return null;
  }
}
